rasterio/_io.pyx

```diff
@@ -1479,6 +1479,8 @@ cdef class RasterUpdater(RasterReader):
 
         # Normally, we expect a CRS dict.
         if isinstance(crs, dict):
+            crs = CRS(crs)
+        if isinstance(crs, CRS):
```

Comment:
Well, the two conditionals are not mutually exclusive, since we modify the `crs` object in the first block if it comes in as a dict. Then we *also* proceed to do everything in the next block, because now it *is* an instance of `CRS`. We do prefer `CRS` object, but we're saying that if it comes in as a `dict`, make it a `CRS` object and then proceed as normal. If we flipped the conditionals, then nothing in the second conditional would get executed when a `dict` comes in, but we want it to be.